testsuite/forge/src/backend/local.rs

```diff
@@ -217,8 +235,25 @@ impl Node for DiemNode {
         unimplemented!()
     }
 
-    fn health_check(&mut self) -> HealthStatus {
-        self.health_check()
+    fn health_check(&mut self) -> Result<(), HealthCheckError> {
+        let mut check_result = Ok(());
+        for _i in 0..HEALTH_CHECK_RETRY {
+            match self.health_check() {
+                HealthStatus::Stopped => {
+                    check_result = Err(HealthCheckError::NotStarted);
+                },
+                HealthStatus::Crashed(e) => {
+                    check_result = Err(HealthCheckError::Crashed(e))
+                },
+                HealthStatus::RpcFailure(e) => {
+                    check_result = Err(HealthCheckError::RpcFailure(e))
+                },
+                _ => return Ok(())
+            };
+            std::thread::sleep(Duration::from_millis(1000));
+        }
+
+        check_result
```

Comment:
Lets change this to not loop, but rather only do the health check once. If we want to have a `wait_for_startup` or similar function that does this waiting we can add that, but the health_check function should probably remain simple.

Reply:
make sense, I will add loop in test case itself